### DISCUSSION TITLE
HttpTransferRequest Error Reason

### DISCUSSION THREAD

**Discussion by henda79** [shinyorg/shiny#1224] [Q&A] [answered]
Hey guys, I'm using `HttpTransferRequest` to upload images on android and I'm subscribing to the `WhenUpdateReceived` method.

When `HttpTransferResult.Status == HttpTransferState.Error` , I can't see a way to get the actual error.

Also, when the transfer fails for whatever reason, I want to re-queue the request, what's the best way to go about that ?

Thanks !

**Comment by aritchie** [comment, marked as answer]
The HttpTransferDelegate returns with the exception in the OnError

**Reply by henda79** [reply]
The OnError does not seem to raise unless via an exception in HttpTransferResult, I've got BP's set if the error returns a 400 Bad Request, the state is set to error but OnError does not raise. Here is my usage.

```
                    var manager = Host.Current.Services.GetRequiredService<IHttpTransferManager>();
                    manager.WhenUpdateReceived().SubscribeAsync(result =>
                        {
                            if (result.Status == HttpTransferState.Error)
                            {
                                return Task.Run(async () =>
                                {
                                    await manager.Cancel(result.Request.Identifier);
                                    await manager.Queue(r);
                                });
                            }

                            return Task.CompletedTask;
                        },
                        //OnError
                        (ex) =>
                        {
                            
                        },
                        //OnComplete
                        () =>
                        {

                        });
```

**Reply by aritchie** [reply]
That's the foreground code for monitoring metrics.  Use the IHttpTransferDelegate if you want to monitor errors.

**Reply by henda79** [reply]
Thanks, I'll have a look.
